readline/plan.sh

```diff
@@ -47,10 +48,14 @@ do_download() {
 
   # Download all patch files, providing the corresponding shasums so we can
   # skip re-downloading if already present and verified
-  for i in $(seq 0 $((${#_patch_files[@]} - 1))); do
-    p="${_patch_files[$i]}"
-    download_file "$p" "$(basename "$p")" "${_patch_shasums[$i]}"
-  done; unset i p
+  if curl -Ifs "${_url_base}/readline-${_base_version}-patches"; then
+    for i in $(seq 0 $((${#_patch_files[@]} - 1))); do
+      p="${_patch_files[$i]}"
+      download_file "$p" "$(basename "$p")" "${_patch_shasums[$i]}"
+    done; unset i p
+  else
+    build_line "No patches directory for ${pkg_name} #{_base_version}"
```

Comment:
Version 8 has no patches yet. I'm a little undecided between making the patching code handle the case of no patches being available (what this PR does) or simply stripping out the patching code (neater but may cause repeated work when readline 8 does get patches and someone needs to enable them).

Reply:
I think it's best to leave in.

I _think_ by using curl you might end up with some odd failure cases that won't do what we expectt though.   Rather than a curl to see if the directory is present, I'd check the length of $patch_files.  I think even that may not be necessary, the `for.. seq` should handle the empty case.

Reply:
You are right. I added the curl bit to handle the immediate error of not being able to get the patch files before I made the change to make the array of current patches empty. I will straighten out the curl.